File: binary_search.py
def binary_search(list, target):
    first = 0
    last = len(list) - 1

    while first <= last:
        midpoint = (first + last) // 2

        if list[midpoint] == target:
            return midpoint
        elif list[midpoint] < target:
            first = midpoint + 1
        else:
            last = midpoint -1
    return 0

def verify(index):
    if index != 0:
       return print("target found at the index: ", index)
    return print("target not found")

list = [1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17]
index= binary_search(list,12)
index2= binary_search(list,19)
verify(index)
verify(index2)

File: binarySearch.js
function binarySearch(list, target) {
    let midpoint;
    first = 0
    last = list.length - 1
    while (first <= last) {
        midpoint = Math.floor((first + last) / 2)
        if (list[midpoint] == target) {
            return midpoint
        } else if (list[midpoint] < target) {
            first = midpoint + 1
        } else if (list[midpoint] > target) {
            last = midpoint - 1
        }
    }
    return null
}

function verify(index) {
    if (index != null) {
        return console.log(`Target found in index:`, index)
    } else {
        return console.log("Target not found!")
    }
}


const list = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]
let index = binarySearch(list, 1)
verify(index)